### PR TITLE
BCI Java: subprocesses

### PR DESCRIPTION
**Pull request by ilausuch**
https://progress.opensuse.org/issues/110689
VR: http://marvin5.arch.suse.cz/tests/961

### CODE REVIEW

File: tests/trainers/java/subprocesses.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  {
+    try {
+        Process process = Runtime.getRuntime().exec(new String[] { "ls", "/" });
+        process.waitFor();
+
+        assert process.exitValue() == 0;
+        
+        OutputStream out = process.getOutputStream();
+
+        InputStreamReader isr = new InputStreamReader(process.getInputStream());
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/trainers/java/subprocesses.java:20]: Good point, this worked perfectly

[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/trainers/java/subprocesses.java]: Thanks!

File: tests/test_openjdk.py [hunk 2]
```diff
@@ -67,28 +67,35 @@ def test_jdk_version(container, java_version):
     indirect=["container_per_test"],
 )
 @pytest.mark.parametrize(
-    "test_to_run, expected_strings, expected_exit_status, java_params",
+    "test_to_run, expected_strings, expected_exit_status, java_params, arguments, environment",
```
Comment (dcermak) [inline tests/test_openjdk.py:70]: Why are you adding additional parameters that you are not even using?

**Review by dcermak — CHANGES_REQUESTED** [review]
Please change the way the parameters are passed to `test_jdk_extended`. I would suggest to use a [`dataclass`](https://docs.python.org/3/library/dataclasses.html) instead of 6 parameters (most of which are unused…).

File: tests/test_openjdk.py (hunk 2 above)
Comment (ilausuch) [reply tests/test_openjdk.py:70]: I will wait for the resolution in https://github.com/SUSE/BCI-tests/pull/157

Comment (ilausuch) [reply tests/test_openjdk.py]: Once #157 the same solution has being used here

File: tests/trainers/java/subprocesses.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        process.waitFor();
+
+        assert process.exitValue() == 0;
+
+        Scanner scanner = new Scanner(process.getInputStream());
+        while (scanner.hasNextLine()) {
+            System.out.println(scanner.nextLine());
+        }
+        scanner.close();
+    }catch(Exception ex) {
```
Comment (grisu48) [inline tests/trainers/java/subprocesses.java:24]:
```suggestion
    } catch(Exception ex) {
```

File: tests/trainers/java/subprocesses.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/**
+Test subprocesses
+Executes a subprocess and check the exit value
+and the results
+*/
+import java.io.*;
+import java.util.Scanner;
+
+public class MemoryTest
```
Comment (grisu48) [inline tests/trainers/java/subprocesses.java:9]: Really MemoryTest? :slightly_smiling_face:

File: tests/test_openjdk.py [hunk 5]
```diff
@@ -120,6 +120,10 @@ class TestExtendedParams:
                 expected_strings=["test"], environment="ENV1=test"
             ),
         ),
+        (
+            "subprocesses",
+            TestExtendedParams(expected_strings=["tmp", "usr"]),
```
Comment (grisu48) [inline tests/test_openjdk.py:125]: Just to be sure: this wants `tmp` and `usr` to be present in the captures output of the command, right?

File: tests/trainers/java/subprocesses.java (hunk 4 above)
Comment (ilausuch) [reply tests/trainers/java/subprocesses.java:9]: :) nice catch

Comment (ilausuch) [reply tests/trainers/java/subprocesses.java]: Done

File: tests/trainers/java/subprocesses.java (hunk 3 above)
Comment (ilausuch) [reply tests/trainers/java/subprocesses.java]: Done

File: tests/trainers/java/subprocesses.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        Scanner scanner = new Scanner(process.getInputStream());
+        while (scanner.hasNextLine()) {
+            System.out.println(scanner.nextLine());
+        }
+        scanner.close();
+    }catch(Exception ex) {
+        System.exit(1);
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/trainers/java/subprocesses.java]: Done

File: tests/test_openjdk.py (hunk 5 above)
Comment (ilausuch) [reply tests/test_openjdk.py:125]: yes 👍

**Review by dcermak — APPROVED** [review]
LGTM

**Pull request merged**